src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -843,6 +843,17 @@ abstract class BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
       }
     }
 
+    /*
+     * must-single-thread
+     */
+    def emitParamNames(jmethod: asm.MethodVisitor, params: List[ValDef]) = {
+      for (param <- params) {
+        var access = asm.Opcodes.ACC_FINAL
+        if (param.mods.isSynthetic)
+          access |= asm.Opcodes.ACC_SYNTHETIC
```

Comment:
To summarize: my suggestion here is to change the above condition to `if (param.symbol.isArtifact)`. The consequence would be that params synthesized by `newSyntheticValueParams` would not be marked synthetic in the bytecode, which is in line with what we are doing in many other cases (e.g. delambdafy methods).